flang/lib/Semantics/check-select-type.cpp

```diff
@@ -258,6 +258,10 @@ void SelectTypeChecker::Enter(const parser::SelectTypeConstruct &construct) {
     if (IsProcedure(*selector)) {
       context_.Say(
           selectTypeStmt.source, "Selector may not be a procedure"_err_en_US);
+    } else if (evaluate::IsAssumedRank(*selector)) {
+      context_.Say(selectTypeStmt.source,
+          "Assumed-rank variable "
```

Comment:
I recommend that you not split error messages across multiple lines -- that makes it much more difficult to grep the sources to find the code that emits a particular error.